Doc/reference/datamodel.rst

```diff
@@ -2376,7 +2376,7 @@ left undefined.
 
    .. note::
 
-      If the right operand's type is a subclass of the left operand's type and that
+      If the right operand's type is a strict subclass of the left operand's type and that
```

Comment:
As @pochmann noted, the problem is actually with the main paragraph.  Instead of

> These functions are only called if the left operand does not support the corresponding operation 3 and the operands are of different types. 4

It should say:

    These functions are only called if:

    - the operands are of different types; and
      - the right operand is a subclass of the left operand 4; or
      - the left operand does not support the corresponding operation 3 

The line with the example just needs

    or `y` is a subclass of `x`

added at the end.

Reply:
Rather than

    or `y` is a subclass of `x`

Did you mean 

    or ``type(y)`` is a subclass of ``type(x)``